Sources/Workspace/Workspace.swift

```diff
@@ -256,6 +256,8 @@ public class Workspace {
     fileprivate var resolvedFileWatcher: ResolvedFileWatcher?
 
     fileprivate let additionalFileRules: [FileRuleDescription]
+    
+    fileprivate let diagnostics: DiagnosticsEngine?
```

Comment:
Is this for workspace-wide diagnostics?  When does it get cleared?  Should this not be done as part of package loading?

Reply:
Good question about clearing, I have to figure that out still. This was mainly needed for `RepositoryPackageContainerProvider` which can load manifests and is instantiated as part of initializing a `Workspace`. 

I think we could get away with not doing this at all as part of this PR, since remote packages won't emit this diagnostic, but it might be not be obvious for future uses of the `DiagnosticsEngine` that's now available during package loading.

Reply:
Right, I think that would be a bit dangerous.  If the diagnostics engine is there, people will add things to it, and it should work correctly.